src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/Wedge.cpp

```diff
@@ -5,139 +5,416 @@
+                            this](const Axis axis) -> T {
+    const size_t axis_idx = axis_index(axis);
+    const double factor = axis_sgn(axis) * outer_surface_.half_cube_length -
+                          gsl::at(projection_center_, axis_idx);
+    T result{};
+    // If we were given an axis to this function, then we can be confident that
+    // all the coords of this axis aren't zero, so we don't have to check them
+    if (potential_axis.has_value()) {
+      result = factor / gsl::at(centered_coords, axis_idx);
+    } else {
+      // However, if we have to figure out the axis, it's possible some coords
+      // for this axis are zero but some aren't so we have to check them all
+      // individually to avoid dividing by zero. If a coordinate is zero, then
+      // that point can't exist in the wedge whose coordinate is zero. Since we
+      // take the min of all positive lambdas to find the correct one, we
+      // overwrite this lambda with -max so it'll never be chosen
+      result = make_with_value<T>(centered_coords[0], factor);
+      for (size_t i = 0; i < get_size(result); i++) {
+        if (get_element(gsl::at(centered_coords, axis_idx), i) == 0.0) {
+          get_element(result, i) = -std::numeric_limits<double>::max();
+        } else {
+          get_element(result, i) /=
+              get_element(gsl::at(centered_coords, axis_idx), i);
+        }
+      }
+    }
+
+    return result;
+  };
+
+  // If we specified an axis that we are on, then use that lambda. This saves
+  // some computation.
+  if (potential_axis.has_value()) {
+    return lambda_axis(potential_axis.value());
+  }
+
+  // Otherwise we need to loop over all of them
+  const std::array all_axes{Axis::PlusZ,  Axis::MinusZ, Axis::PlusY,
+                            Axis::MinusY, Axis::PlusX,  Axis::MinusX};
+  std::array<T, 6> candidate_lambdas{};
+
+  for (size_t i = 0; i < all_axes.size(); i++) {
+    gsl::at(candidate_lambdas, i) = lambda_axis(gsl::at(all_axes, i));
+  }
+
+  // We need to take the min of all the positive lambdas
+  const double fill_value = std::numeric_limits<double>::max();
+  T result = make_with_value<T>(candidate_lambdas[0], fill_value);
+  std::vector<Axis> chosen_axes(get_size(result), Axis::None);
+  for (size_t i = 0; i < get_size(result); i++) {
+    for (size_t j = 0; j < candidate_lambdas.size(); j++) {
+      if (get_element(gsl::at(candidate_lambdas, j), i) > 0.0 and
+          get_element(gsl::at(candidate_lambdas, j), i) <
+              get_element(result, i)) {
+        get_element(result, i) = get_element(gsl::at(candidate_lambdas, j), i);
+        chosen_axes[i] = gsl::at(all_axes, j);
+      }
+    }
+  }
+
+  ASSERT(not alg::any_of(chosen_axes,
+                         [&](const Axis& a) { return a == Axis::None; }),
+         "No candidate cube lambdas in the Wedge shape map transition function "
+         "are negative. This is an internal error. Please file an issue.");
```

Comment:
Can you print some info for debugging out? I've been trying to add more info to our errors